centreon/www/class/centreonLog.class.php

```diff
@@ -129,77 +147,379 @@ public static function singleton($uid = 0)
+    {
+        $this->log($type, self::LEVEL_INFO, $message, $customContext, $exception);
+    }
+
+    /**
+     * @param int $type TYPE_* constants
+     * @param string $message
+     * @param array $customContext
+     * @param Throwable|null $exception
+     * @return void
+     */
+    public function warning(int $type, string $message, array $customContext = [], ?Throwable $exception = null): void
+    {
+        $this->log($type, self::LEVEL_WARNING, $message, $customContext, $exception);
+    }
+
+    /**
+     * @param int $type TYPE_* constants
+     * @param string $message
+     * @param array $customContext
+     * @param Throwable|null $exception
+     * @return void
+     */
+    public function error(int $type, string $message, array $customContext = [], ?Throwable $exception = null): void
+    {
+        $this->log($type, self::LEVEL_ERROR, $message, $customContext, $exception);
+    }
+
+    /**
+     * @param int $type TYPE_* constants
+     * @param string $message
+     * @param array $customContext
+     * @param Throwable|null $exception
+     * @return void
+     */
+    public function critical(int $type, string $message, array $customContext = [], ?Throwable $exception = null): void
+    {
+        $this->log($type, self::LEVEL_CRITICAL, $message, $customContext, $exception);
+    }
+
+    /**
+     * @param int $type TYPE_* constants
+     * @param string $message
+     * @param array $customContext
+     * @param Throwable|null $exception
+     * @return void
+     */
+    public function alert(int $type, string $message, array $customContext = [], ?Throwable $exception = null): void
+    {
+        $this->log($type, self::LEVEL_ALERT, $message, $customContext, $exception);
+    }
+
+    /**
+     * @param int $type TYPE_* constants
+     * @param string $message
+     * @param array $customContext
+     * @param Throwable|null $exception
+     * @return void
+     */
+    public function emergency(int $type, string $message, array $customContext = [], ?Throwable $exception = null): void
+    {
+        $this->log($type, self::LEVEL_EMERGENCY, $message, $customContext, $exception);
+    }
+
+    /**
+     * @return array
+     */
+    public function getLogFileHandler(): array
+    {
+        return $this->logFileHandler;
+    }
+
+    /**
+     * @param int $logTypeId
+     * @param string $logFileName
+     * @return CentreonLog
+     */
+    public function pushLogFileHandler(int $logTypeId, string $logFileName): CentreonLog
+    {
+        $pathLogFileName = '';
+        $logFile = '';
+        $explodeFineName = explode(DIRECTORY_SEPARATOR, $logFileName);
+        if (! empty($explodeFineName)) {
+            $logFile = $explodeFineName[count($explodeFineName) - 1];
+            unset($explodeFineName[count($explodeFineName) - 1]);
+            $pathLogFileName = implode(DIRECTORY_SEPARATOR, $explodeFineName);
```

Comment:
Typo: I guess you meant `explodeFileName`

Reply:
fixed on PR #4862